exec/cnex/array.c

```diff
@@ -45,13 +53,17 @@ Array *array_expandArray(Array *self, size_t iElements)
         if (self == NULL) {
             fatal_error("Unable to allocate array for array expansion.");
         }
-        self->size = 0;
-        self->data = NULL;
+        self->size = iElements;
+        self->max = ARRAY_INITIAL_SIZE;
+        self->data = malloc(sizeof(Cell) * self->max);
     }
 
-    self->data = realloc(self->data, sizeof(Cell) * (self->size + iElements));
-    if (self->data == NULL) {
-        fatal_error("Unable to allocate memory for %d expanded array elements.", self->size + iElements);
+    if (self->size + iElements > self->max) {
+        self->max = self->size + iElements;
```

Comment:
Isn't this where you want to double the allocation size? If you were to call `array_expandArray()` one element at a time, you'd be back to where you started as far as reallocating and copying.